packages/material-ui-based-components/src/components/Wizard.tsx

```diff
@@ -0,0 +1,122 @@
+import React, { useContext } from 'react';
+import Box from '@mui/material/Box';
+import Stepper from '@mui/material/Stepper';
+import Step from '@mui/material/Step';
+import StepLabel from '@mui/material/StepLabel';
+import Button from '@mui/material/Button';
+import Typography from '@mui/material/Typography';
+import { withRuleEnginePanel } from '../shared/withRuleEngine';
+import { PROPS_PANEL } from '../utils/types';
+import { FormContext, getRenderer } from '@aemforms/af-react-renderer';
+
+
+interface TabPanelProps {
+    children?: any;
+    index: number;
+    value: number;
+}
+
+function TabPanel(props: TabPanelProps) {
+    const { children, value, index, ...other } = props;
+
+    return (
+        <div
+            hidden={value !== index}
+            id={children.id}
+            {...other}
+        >
+            {value === index && (
+                <Box sx={{ p: 3 }}>
+                    {children}
+                </Box>
+            )}
+        </div>
+    );
+}
+
+function Wizard(props: PROPS_PANEL) {
+    //@ts-ignore
+    const mappings = useContext(FormContext).mappings;
+    const items = props.items || [];
+    const [activeStep, setActiveStep] = React.useState(0);
+    const [value, setValue] = React.useState(0);
+
+    const steps = items.map((child: any) => (
+        child.label.value
+    ));
+
+    const getChild = (child: any, index: number) => {
+        const Comp = getRenderer(child, mappings);
+        return Comp ? <Comp key={`${child.id}_${index}`} {...child} /> : (null);
+    };
+
+    const isStepOptional = (step: number) => {
```

Comment:
bind with useCallback method

Reply:
Binded with callback

Reply:
isStepOptional method is removed since it hardcoded the step which was optional and we can basically send the field to be required. Thus, this method became unnecessary.